TemplateSrv: Formatting options for ${__from} and ${__to}, unix seconds epoch, ISO 8601/RFC 3339 (#26466)
grafana/grafana: docs/sources/variables/global-variables.md

# Global variables

> Grafana has global built-in variables that can be used in expressions in the query editor. This topic lists them in alphabetical order and defines them. These variables are useful in both queries, dashboard links, panel links & data links.

Reviewer: ```suggestion
Grafana has global built-in variables that can be used in expressions in the query editor. This topic lists them in alphabetical order and defines them. These variables are useful in queries, dashboard links, panel links, and data links.
```

Reviewer: "Both" means you only have two items in your list. Also, please do not use ampersands as an abbreviation for "and" in prose.

## $__dashboard

- `${__dashboard.name}` is the name of the current dashboard
- `${__dashboard.uid}` is the UID of the current dashboard (used in url)

## $__from and $__to

> Grafana has two built in time range variables: `$__from` and `$__to`. They are currently always interpolated as epoch milliseconds by default but you can control date formatting.

Reviewer: ```suggestion
Grafana has two built-in time range variables: `$__from` and `$__to`. By default, they areinterpolated as epoch milliseconds, but you can change date formatting.
```

> This special formatting syntax is only available in Grafan a 7.1.2+

| Syntax                   | Example result           | Description |
| ------------------------ | ------------------------ | ----------- |
| `${__from}`              | 1594671549254            | Unix millisecond epoch |
| `${__from:date}`         | 2020-07-13T20:19:09.254Z | No args, defaults to ISO 8601/RFC 3339 | 
| `${__from:date:iso}`     | 2020-07-13T20:19:09.254Z | ISO 8601/RFC 3339 |
| `${__from:date:seconds}` | 1594671549               | Unix seconds epoch |
| `${__from:date:YYYY-MM}` | 2020-07                  | Any custom [date format](https://momentjs.com/docs/#/displaying/) |

> The above syntax of course works with `${__to}` as well.  

Reviewer: ```suggestion
The above syntax oworks with `${__to}` as well.  
```

## $__interval

> The `$__interval` variable can be used as a parameter to group by time (for InfluxDB, MySQL, Postgres, MSSQL), Date histogram interval (for Elasticsearch) or as a _summarize_ function parameter (for Graphite).

Reviewer: ```suggestion
You can use the `$__interval` variable as a parameter to group by time (for InfluxDB, MySQL, Postgres, MSSQL), Date histogram interval (for Elasticsearch), or as a _summarize_ function parameter (for Graphite).
```

## $__range

Currently only supported for Prometheus data sources. This variable represents the range for the current dashboard. It is calculated by `to - from`. It has a millisecond and a second representation called `$__range_ms` and `$__range_s`.

## $timeFilter or $__timeFilter

The `$timeFilter` variable returns the currently selected time range as an expression. For example, the time range interval `Last 7 days` expression is `time > now() - 7d`.

This is used in several places, including:

> - The WHERE clause for the InfluxDB data source. Grafana adds it automatically to InfluxDB queries when in Query Editor Mode. It has to be added manually in Text Editor Mode: `WHERE $timeFilter`.

Reviewer: ```suggestion
- The WHERE clause for the InfluxDB data source. Grafana adds it automatically to InfluxDB queries when in Query Editor mode. You can add it manually in Text Editor mode: `WHERE $timeFilter`.
```
